### PR TITLE
Refactor admin/customer_details controller to use order contents

### PR DESCRIPTION
**Pull request by athal7**
* Bring over updates and specs from master for that controller.
* Introduce OrderContents#associate user (soon to become beefier to pull it out of the order model).
* Use OrderContents from the customer_details controller.

### CODE REVIEW

**Comment by jordan-brough** [comment]
nice!

File: backend/app/controllers/spree/admin/orders/customer_details_controller.rb [hunk 1]
```diff
@@ -2,34 +2,34 @@ module Spree
...
+          if @order.contents.update_cart(order_params)
             if params[:guest_checkout] == "false"
-              @order.associate_user!(Spree.user_class.find(params[:user_id]), @order.email.blank?)
+              requsted_user = Spree.user_class.find(params[:user_id])
+              @order.contents.associate_user(requsted_user, @order.email.blank?)
             end
-            while @order.next; end
-
+            @order.next
             @order.refresh_shipment_rates
```
Comment (jordan-brough) [inline backend/app/controllers/spree/admin/orders/customer_details_controller.rb]: We should get `refresh_shipment_rates` into OrderContents as well right?  You going to do that in a separate PR?

Comment (jordan-brough) [inline backend/app/controllers/spree/admin/orders/customer_details_controller.rb]: Also `order.next` yeah?  Separate PR there also?

Comment (athal7) [inline backend/app/controllers/spree/admin/orders/customer_details_controller.rb]: so my thoughts about those are that they ultimately shouldn't need to be called here. they aren't doing anything material to change the makeup of the order. ultimately IMHO the 'next'-ing and refreshing of anything ought to be taken care of by the 'updater' that we use to surround our individual content changes, and once we figure out how that plays together then we can surround replace those calls here by how the updating works. i can be convinced otherwise though.

Comment (jordan-brough) [inline backend/app/controllers/spree/admin/orders/customer_details_controller.rb]: awesome, i think that's a great plan.

File: backend/app/controllers/spree/admin/orders/customer_details_controller.rb [hunk 2]
```diff
@@ -2,34 +2,34 @@ module Spree
...
           @order.bill_address.country_id = country_id if @order.bill_address.country.nil?
           @order.ship_address.country_id = country_id if @order.ship_address.country.nil?
         end
 
         def update
-          if @order.update_attributes(order_params)
+          if @order.contents.update_cart(order_params)
             if params[:guest_checkout] == "false"
-              @order.associate_user!(Spree.user_class.find(params[:user_id]), @order.email.blank?)
+              requsted_user = Spree.user_class.find(params[:user_id])
```
Comment (jordan-brough) [inline backend/app/controllers/spree/admin/orders/customer_details_controller.rb]: fyi mispelling - s/requsted_user/requested_user/

**Comment by jordan-brough** [comment]
:+1: awesome

File: backend/app/controllers/spree/admin/orders/customer_details_controller.rb (hunk 2 above)
Comment (athal7) [inline backend/app/controllers/spree/admin/orders/customer_details_controller.rb]: derp

**Comment by athal7** [comment]
89f33ba
